tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -342,15 +338,39 @@ internal TracerSettings(IConfigurationSource? source, IConfigurationTelemetry te
             IsRunningInAzureAppService = config
                                         .WithKeys(ConfigurationKeys.AzureAppService.AzureAppServicesContextKey)
                                         .AsBool(false);
-            if (IsRunningInAzureAppService)
+
+            // we only want to instantiate ImmutableAzureAppServiceSettings if we are for sure running
+            // in AAS, or if we might be running in a Consumption Plan function.
+            var isMaybeAzureFunctionConsumptionPlan = config.WithKeys(ConfigurationKeys.AzureAppService.WebsiteSKU).AsString("Dynamic") == "Dynamic";
+
+            if (IsRunningInAzureAppService || isMaybeAzureFunctionConsumptionPlan)
             {
                 AzureAppServiceMetadata = new ImmutableAzureAppServiceSettings(source, _telemetry);
                 if (AzureAppServiceMetadata.IsUnsafeToTrace)
                 {
                     TraceEnabledInternal = false;
                 }
+
+                // IsRunningInAzureFunctionsConsumptionPlan = AzureAppServiceMetadata.IsFunctionsAppConsumptionPlan;
             }
 
+            /*
+            else
+            {
+                IsRunningInAzureFunctionsConsumptionPlan = false;
+            }
+            */
+            IsRunningInAzureFunctionsConsumptionPlan = false;
```

Comment:
OK, maybe this PR isn't ready yet? 😅

Reply:
was wrangling integration tests, ready now :)